project_forecast_line/tests/test_forecast_line.py

```diff
@@ -0,0 +1,875 @@
+            "forecast_role_id": self.role_consultant.id,
+            "forecast_date_planned_start": "2022-02-14",
+            "forecast_date_planned_end": "2022-02-14",
+            "planned_hours": 8,
+        }
+        task_values.update({"name": "Task1"})
+        task_1 = self.env["project.task"].create(task_values)
+        task_1.user_id = self.user_consultant
+        task_values.update({"name": "Task2"})
+        task_2 = self.env["project.task"].create(task_values)
+        task_2.user_id = self.user_consultant
+
+        # Project 2 is in stage "in progress" to get forecast
+        project_2 = self.env["project.project"].create({"name": "TestProject2"})
+        project_2.project_status = self.env.ref(
+            "project_status.project_status_in_progress"
+        )
+        task_values.update({"project_id": project_2.id, "name": "Task3"})
+        task_3 = self.env["project.task"].create(task_values)
+        task_3.user_id = self.user_consultant
+        task_values.update({"name": "Task4"})
+        task_4 = self.env["project.task"].create(task_values)
+        task_4.user_id = self.user_consultant
+
+        # check forecast lines
+        forecast = self.env["forecast.line"].search(
+            [("task_id", "in", (task_1.id, task_2.id, task_3.id, task_4.id))]
+        )
+        self.assertEqual(len(forecast), 4)
+        self.assertEqual(
+            forecast.mapped("forecast_hours"),
+            [
+                -8.0,
+            ]
+            * 4,
+        )
+        # consolidated forecast is in days of 8 hours
+        self.assertEqual(forecast.mapped("consolidated_forecast"), [1.0] * 4)
+        self.assertEqual(
+            forecast.filtered(lambda r: r.type == "forecast").mapped(
+                "confirmed_consolidated_forecast"
+            ),
+            [0.0] * 2,
+        )
+        self.assertEqual(
+            forecast.filtered(lambda r: r.type == "confirmed").mapped(
+                "confirmed_consolidated_forecast"
+            ),
+            [1.0] * 2,
+        )
+        forecast_consultant, forecast_pm = self._get_employee_forecast()
+        self.assertEqual(forecast_consultant.forecast_hours, 6.0)
+        self.assertAlmostEqual(
+            forecast_consultant.consolidated_forecast, 1.0 * 75 / 100 - 4
+        )
+        self.assertAlmostEqual(
+            forecast_consultant.confirmed_consolidated_forecast, 1.0 * 75 / 100 - 2
+        )
+        self.assertEqual(forecast_pm.forecast_hours, 2.0)
+        self.assertAlmostEqual(forecast_pm.consolidated_forecast, 0.25)
+        self.assertAlmostEqual(forecast_pm.confirmed_consolidated_forecast, 0.25)
+
+    @freeze_time("2022-01-01 12:00:00")
+    def test_forecast_with_holidays(self):
+        self.test_task_forecast_lines_consolidated_forecast()
+        with Form(self.env["hr.leave"]) as form:
+            form.employee_id = self.employee_consultant
+            form.holiday_status_id = self.env.ref("hr_holidays.holiday_status_unpaid")
```

Comment:
@ntsirintanis Currently this test is failing on `odoo.exceptions.ValidationError: Unpaid are only valid starting from 2023-01-01`.

I think it's because of [this line](https://github.com/odoo/odoo/blob/14.0/addons/hr_holidays/data/hr_holidays_data.xml#L49), which is removed in Odoo 15.0.

I think we can patch this test right at the beginning by removing or changing `validity_start` of `hr_holidays.holiday_status_unpaid`?